pkg/server/db_options.go

```diff
@@ -59,12 +59,26 @@ type dbOptions struct {
 
 	IndexOptions *indexOptions `json:"indexOptions"`
 
+	Autoload featureState `json:"autoload"` // unspecfied is considered as enabled for backward compatibility
+
 	CreatedBy string    `json:"createdBy"`
 	CreatedAt time.Time `json:"createdAt"`
 	UpdatedBy string    `json:"updatedBy"`
 	UpdatedAt time.Time `json:"updatedAt"`
 }
 
+type featureState int
+
+const (
+	unspecifiedState featureState = 0
+	enabledState     featureState = 1
+	disabledState    featureState = 2
+)
+
+func (fs featureState) isEnabled() bool {
+	return fs == unspecifiedState || fs == enabledState
+}
+
```

Comment:
Maybe add some function to set the state so that concrete values are hidden?
```suggestion

func (fs *featureState) setEnabled(enabled bool) {
    if enabled {
        *fs = enabledState
    } else {
        *fs = disabledState
    }
}
```